src/platform.rs

```diff
@@ -325,13 +339,166 @@ impl VirtualProcessor {
         })?;
         Ok((translation_result, gpa))
     }
+
+    pub fn query_gpa_range_dirty_bitmap(
+        &self,
+        gva: WHV_GUEST_PHYSICAL_ADDRESS,
+        range_size_in_bytes: UINT64,
+        bitmap_size_in_bytes: UINT32,
+    ) -> Result<(Box<[UINT64]>), WHPError> {
+        let num_elem = bitmap_size_in_bytes / std::mem::size_of::<UINT64>() as UINT32;
+        let mut bitmap: Box<[UINT64]> = vec![0; num_elem as usize].into_boxed_slice();
+
+        check_result(unsafe {
+            WHvQueryGpaRangeDirtyBitmap(
+                *self.partition.borrow().handle(),
+                gva,
+                range_size_in_bytes,
+                bitmap.as_mut_ptr(),
+                bitmap_size_in_bytes,
+            )
+        })?;
+        Ok(bitmap)
+    }
+
+    pub fn get_lapic(&self) -> Result<LapicStateRaw, WHPError> {
+        let mut state: LapicStateRaw = Default::default();
+        let mut written_size: UINT32 = 0;
+
+        check_result(unsafe {
+            WHvGetVirtualProcessorInterruptControllerState(
+                *self.partition.borrow().handle(),
+                self.index,
+                &mut state as *mut _ as *mut VOID,
+                std::mem::size_of::<LapicStateRaw>() as UINT32,
+                &mut written_size,
+            )
+        })?;
+        Ok(state)
+    }
+
+    pub fn set_lapic(&self, state: &LapicStateRaw) -> Result<(), WHPError> {
+        check_result(unsafe {
+            WHvSetVirtualProcessorInterruptControllerState(
+                *self.partition.borrow().handle(),
+                self.index,
+                state as *const _ as *const VOID,
+                std::mem::size_of::<LapicStateRaw>() as UINT32,
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn request_interrupt(&self, interrupt: &WHV_INTERRUPT_CONTROL) -> Result<(), WHPError> {
+        check_result(unsafe {
+            WHvRequestInterrupt(
+                *self.partition.borrow_mut().handle(),
+                interrupt,
+                std::mem::size_of::<WHV_INTERRUPT_CONTROL>() as UINT32,
+            )
+        })?;
+        Ok(())
+    }
+
+    #[allow(unreachable_patterns)] // Future-proof against new WHV_PARTITION_COUNTER_SET values
+    pub fn get_partition_counters(
+        &self,
+        partition_counter_set: WHV_PARTITION_COUNTER_SET,
+    ) -> Result<(WHV_PARTITION_COUNTERS), WHPError> {
```

Comment:
WHV_PARTITION_COUNTERS is undefined.

Reply:
"Undefined" errors resolved with addition of missing win_hv_platform_defs_internal.rs.